src/extension.ts

```diff
@@ -99,14 +100,21 @@ function onDidChangeTelemetryEnabled(): Disposable {
 
 async function openTelemetryOptInDialogIfNeeded(context: ExtensionContext) {
   logTelemetryStatus();
-  const optInRequested: boolean | undefined = context.globalState.get(
-    OPT_IN_STATUS_KEY
-  );
+  const optInRequested: boolean | undefined = context.globalState.get(OPT_IN_STATUS_KEY);
+
   Logger.log(`optInRequested is: ${optInRequested}`);
 
   if (optInRequested) {
```

Comment:
Small suggestion, otherwise all looks good. 👍🏼 

```suggestion
if (optInRequested  && retryOptin) {
      clearInterval(retryOptin);
    }
```

Reply:
I'd rather not. while it's unlikely to happen the current logic is more correct. You need to return early if optInRequested == true, regardless of the  state of retryOptin

Reply:
yes that's true. Looks good to me then.